### ISSUE TITLE
wrist_3_joint suddenly stops following trajectory in sim only

### ISSUE THREAD

**Issue by mrjogo** [UniversalRobots/Universal_Robots_ROS2_Ignition_Simulation#64]
ROS version: Jazzy
Universal_Robots_ROS2_GZ_Simulation version: 54a6a11a88b886f886cfa098bb21bf8448ea8c18
OS Version: Ubuntu 24.04
Gazebo version: 8.6.0

I've run into a trajectory that reliably fails at the same spot in Gazebo, but not on the real robot:

![wrist_3_bad_trajectory](https://github.com/user-attachments/assets/7d82ece9-f117-4324-b44b-5d553378a9a4)

![wrist_3_bad_trajectory_working](https://github.com/user-attachments/assets/d68bfa50-1db6-494b-b323-048965b27f2c)

The trajectory (included at the bottom of this issue) is run via `/execute_trajectory` and generates these logs:

```
[move_group-7] [INFO] [1731014172.271527322] [move_group.moveit.moveit.ros.move_group.clear_octomap_service]: Execution request received
[move_group-7] [INFO] [1731014172.271574042] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 4 controllers in list
[move_group-7] [INFO] [1731014172.271587231] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 4 controllers in list
[move_group-7] [INFO] [1731014172.271616562] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Validating trajectory with allowed_start_tolerance 0.01
[move_group-7] [INFO] [1731014172.271664211] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Starting trajectory execution ...
[move_group-7] [INFO] [1731014172.271707534] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 4 controllers in list
[move_group-7] [INFO] [1731014172.271730790] [move_group.moveit.moveit.plugins.simple_controller_manager]: Returned 4 controllers in list
[move_group-7] [INFO] [1731014172.271837976] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: sending trajectory to joint_trajectory_controller
[gazebo-5] [INFO] [1731014172.271977000] [joint_trajectory_controller]: Received new action goal
[gazebo-5] [INFO] [1731014172.271996272] [joint_trajectory_controller]: Accepted new action goal
[move_group-7] [INFO] [1731014172.272071023] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: joint_trajectory_controller started execution
[move_group-7] [INFO] [1731014172.272075595] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: Goal request accepted!
[gazebo-5] [ERROR] [1731014175.702744660] [tolerances]: State tolerances failed for joint 5:
[gazebo-5] [ERROR] [1731014175.702779754] [tolerances]: Position Error: -0.201932, Position Tolerance: 0.200000
[gazebo-5] [WARN] [1731014175.702788180] [joint_trajectory_controller]: Aborted due to state tolerance violation
[move_group-7] [WARN] [1731014175.722281434] [moveit.simple_controller_manager.follow_joint_trajectory_controller_handle]: Controller 'joint_trajectory_controller' failed with error PATH_TOLERANCE_VIOLATED: Aborted due to path tolerance violation
[move_group-7] [WARN] [1731014175.722306237] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Controller handle joint_trajectory_controller reports status ABORTED
[move_group-7] [INFO] [1731014175.722324537] [move_group.moveit.moveit.ros.trajectory_execution_manager]: Completed trajectory execution with status ABORTED ...
[move_group-7] [INFO] [1731014175.722370150] [move_group.moveit.moveit.ros.move_group.clear_octomap_service]: Execution completed: ABORTED
```

Looking at the feedback logs for `wrist_3_joint` (the failing joint), it appears that it just stops following the commanded velocity (note that `wrist_3_joint_feedback/*` appear to be slightly shifted in time from `wrist_3_joint_commanded_trajectory/*`):

![image](https://github.com/user-attachments/assets/cac037d1-eb47-47e1-971f-30613cf4e354)

Let me know if you need additional information.

<details>

<summary>Here is the `RobotTrajectory` YAML</summary>

```yaml
joint_trajectory:
  header:
    stamp:
      sec: 0
      nanosec: 0
    frame_id: "world"
  joint_names:
  - "shoulder_pan_joint"
  - "shoulder_lift_joint"
  - "elbow_joint"
  - "wrist_1_joint"
  - "wrist_2_joint"
  - "wrist_3_joint"
  points:
  -
    positions:
    - 1.25078
    - -1.46551
    - 2.01958
    - -0.555325
    - 1.25187
    - -1.57071
    velocities:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    accelerations:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    effort: []
    time_from_start:
      sec: 0
      nanosec: 0
  -
    positions:
    - 1.23783
    - -1.46567
    - 2.01954
    - -0.555365
    - 1.23904
    - -1.61071
    velocities:
    - -0.258989
    - -0.00316612
    - -0.000775924
    - -0.000783988
    - -0.256596
    - -0.800000
    accelerations:
    - -2.58989
    - -0.0316612
    - -0.00775924
    - -0.00783988
    - -2.56596
    - -8.00000
    effort: []
    time_from_start:
      sec: 0
      nanosec: 100000000
  -
    positions:
    - 1.19911
    - -1.46558
    - 2.01856
    - -0.554829
    - 1.20045
    - -1.73071
    velocities:
    - -0.515870
    - 0.00996849
    - -0.0248413
    - 0.0125849
    - -0.515884
    - -1.60796
    accelerations:
    - -2.45326
    - 0.305148
    - -0.413544
    - 0.0962743
    - -2.45306
    - -8.00037
    effort: []
    time_from_start:
      sec: 0
      nanosec: 200000000
  -
    positions:
    - 1.13752
    - -1.46271
    - 2.01335
    - -0.552776
    - 1.13885
    - -1.93071
    velocities:
    - -0.619002
    - 0.0393575
    - -0.0855904
    - 0.0432655
    - -0.619044
    - -2.40640
    accelerations:
    - 2.90212
    - -0.0872223
    - -0.776213
    - 0.873199
    - 2.90006
    - -7.99991
    effort: []
    time_from_start:
      sec: 0
      nanosec: 300000000
  -
    positions:
    - 1.07603
    - -1.45716
    - 2.00202
    - -0.547305
    - 1.07736
    - -2.21050
    velocities:
    - -0.536185
    - 0.0627170
    - -0.124322
    - 0.0587852
    - -0.536230
    - -3.14159
    accelerations:
    - 3.91497
    - -0.142480
    - 0.529087
    - -0.366881
    - 3.91561
    - -1.71201e-10
    effort: []
    time_from_start:
      sec: 0
      nanosec: 400000000
  -
    positions:
    - 1.04112
    - -1.45189
    - 1.99357
    - -0.544316
    - 1.04245
    - -2.52244
    velocities:
    - -0.148465
    - 0.0391928
    - -0.0261299
    - -0.0138536
    - -0.148381
    - -2.95291
    accelerations:
    - 4.03532
    - -0.341215
    - 1.41253
    - -1.04843
    - 4.03719
    - 7.99349
    effort: []
    time_from_start:
      sec: 0
      nanosec: 500000000
  -
    positions:
    - 1.04183
    - -1.44902
    - 1.99695
    - -0.550548
    - 1.04318
    - -2.77778
    velocities:
    - 0.124524
    - 0.0234319
    - 0.0810593
    - -0.103700
    - 0.124785
    - -2.15303
    accelerations:
    - 1.63729
    - -0.0286136
    - 0.661818
    - -0.623791
    - 1.63860
    - 7.99929
    effort: []
    time_from_start:
      sec: 0
      nanosec: 600000000
  -
    positions:
    - 1.05893
    - -1.44659
    - 2.00672
    - -0.562652
    - 1.06031
    - -2.95307
    velocities:
    - 0.187353
    - 0.0253872
    - 0.100197
    - -0.124466
    - 0.187658
    - -1.35269
    accelerations:
    - -0.216800
    - -0.0127023
    - -0.202388
    - 0.213385
    - -0.217316
    - 8.00196
    effort: []
    time_from_start:
      sec: 0
      nanosec: 700000000
  -
    positions:
    - 1.07412
    - -1.44511
    - 2.01378
    - -0.571096
    - 1.07552
    - -3.05053
    velocities:
    - 0.0812026
    - -0.0566741
    - -0.0309924
    - 0.0879889
    - 0.0809308
    - -0.742434
    accelerations:
    - -2.40222
    - -3.50992
    - -4.50892
    - 7.99803
    - -2.42445
    - 0.427262
    effort: []
    time_from_start:
      sec: 0
      nanosec: 800000000
  -
    positions:
    - 1.07662
    - -1.46356
    - 1.99499
    - -0.533878
    - 1.07791
    - -3.14111
    velocities:
    - 0.0355694
    - -0.274810
    - -0.295445
    - 0.569846
    - 0.0339206
    - -1.27166
    accelerations:
    - 0.751959
    - -1.52161
    - -1.99967
    - 3.52191
    - 0.741300
    - -7.99990
    effort: []
    time_from_start:
      sec: 0
      nanosec: 900000000
  -
    positions:
    - 1.08581
    - -1.49764
    - 1.95524
    - -0.460078
    - 1.08688
    - -3.30766
    velocities:
    - 0.170997
    - -0.396859
    - -0.504163
    - 0.901064
    - 0.168298
    - -2.06577
    accelerations:
    - 2.06826
    - -0.868367
    - -2.18866
    - 3.06618
    - 2.05802
    - -7.99925
    effort: []
    time_from_start:
      sec: 1
      nanosec: 0
  -
    positions:
    - 1.11626
    - -1.53986
    - 1.89389
    - -0.356445
    - 1.11701
    - -3.55420
    velocities:
    - 0.473460
    - -0.426521
    - -0.722416
    - 1.15052
    - 0.469879
    - -2.87228
    accelerations:
    - 3.99919
    - 0.370503
    - -2.06530
    - 1.71641
    - 3.99262
    - -7.99724
    effort: []
    time_from_start:
      sec: 1
      nanosec: 100000000
  -
    positions:
    - 1.18225
    - -1.57641
    - 1.81676
    - -0.242496
    - 1.18264
    - -3.86359
    velocities:
    - 0.821558
    - -0.268698
    - -0.766020
    - 1.03830
    - 0.818268
    - -3.14159
    accelerations:
    - 2.87024
    - 2.12485
    - 0.572236
    - -2.68121
    - 2.87957
    - -5.83679e-10
    effort: []
    time_from_start:
      sec: 1
      nanosec: 200000000
  -
    positions:
    - 1.27666
    - -1.59262
    - 1.74438
    - -0.153478
    - 1.27678
    - -4.17494
    velocities:
    - 1.00067
    - -0.0521754
    - -0.638277
    - 0.695013
    - 0.998514
    - -2.92809
    accelerations:
    - -1.00622
    - 2.01276
    - 2.76429
    - -4.78162
    - -0.990054
    - 7.99833
    effort: []
    time_from_start:
      sec: 1
      nanosec: 300000000
  -
    positions:
    - 1.36889
    - -1.59028
    - 1.69329
    - -0.104314
    - 1.36886
    - -4.42789
    velocities:
    - 0.816261
    - 0.0763873
    - -0.391883
    - 0.319195
    - 0.815370
    - -2.12799
    accelerations:
    - -2.63127
    - 0.620551
    - 2.20096
    - -2.83376
    - -2.62191
    - 7.99920
    effort: []
    time_from_start:
      sec: 1
      nanosec: 400000000
  -
    positions:
    - 1.43647
    - -1.58115
    - 1.66388
    - -0.0837274
    - 1.43639
    - -4.60083
    velocities:
    - 0.525552
    - 0.0918761
    - -0.206066
    - 0.116557
    - 0.525301
    - -1.32797
    accelerations:
    - -3.13732
    - -0.237155
    - 1.56932
    - -1.34638
    - -3.13342
    - 7.99941
    effort: []
    time_from_start:
      sec: 1
      nanosec: 500000000
  -
    positions:
    - 1.47332
    - -1.57401
    - 1.65020
    - -0.0770192
    - 1.47322
    - -4.69377
    velocities:
    - 0.209538
    - 0.0429678
    - -0.0752541
    - 0.0332283
    - 0.209487
    - -0.528000
    accelerations:
    - -3.17482
    - -0.651028
    - 1.14021
    - -0.503459
    - -3.17404
    - 8.00000
    effort: []
    time_from_start:
      sec: 1
      nanosec: 600000000
  -
    positions:
    - 1.48027
    - -1.57259
    - 1.64770
    - -0.0759168
    - 1.48017
    - -4.71129
    velocities:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    accelerations:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    effort: []
    time_from_start:
      sec: 1
      nanosec: 666176828
  -
    positions:
    - 1.48027
    - -1.57259
    - 1.64770
    - -0.0759168
    - 1.48017
    - -4.71129
    velocities:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    accelerations:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    effort: []
    time_from_start:
      sec: 1
      nanosec: 666276828
  -
    positions:
    - 1.48033
    - -1.56836
    - 1.68348
    - -0.115917
    - 1.48036
    - -4.71134
    velocities:
    - 0.00120962
    - 0.0851757
    - 0.714950
    - -0.800027
    - 0.00376326
    - -0.000987743
    accelerations:
    - 0.0103694
    - 1.45805
    - 6.54668
    - -8.00365
    - 0.0356832
    - -0.0110825
    effort: []
    time_from_start:
      sec: 1
      nanosec: 766276828
  -
    positions:
    - 1.48044
    - -1.55735
    - 1.74975
    - -0.193191
    - 1.48071
    - -4.71144
    velocities:
    - 0.000559163
    - 0.0750260
    - 0.350959
    - -0.425928
    - 0.00190727
    - -0.000584687
    accelerations:
    - -0.0105025
    - -1.40918
    - -6.59190
    - 8.00000
    - -0.0358233
    - 0.0109819
    effort: []
    time_from_start:
      sec: 1
      nanosec: 866276828
  -
    positions:
    - 1.48045
    - -1.55686
    - 1.75411
    - -0.198036
    - 1.48074
    - -4.71144
    velocities:
    - -0.000432163
    - -0.0605451
    - -0.272657
    - 0.333157
    - -0.00148568
    - 0.000460969
    accelerations:
    - -0.0106430
    - -1.35788
    - -6.64185
    - 7.99867
    - -0.0359807
    - 0.0108791
    effort: []
    time_from_start:
      sec: 1
      nanosec: 966276828
  -
    positions:
    - 1.48035
    - -1.56698
    - 1.69750
    - -0.131319
    - 1.48043
    - -4.71136
    velocities:
    - -0.00101124
    - -0.0834058
    - -0.605054
    - 0.688373
    - -0.00320522
    - 0.000870079
    accelerations:
    - 0.0104621
    - 1.41146
    - 6.59263
    - -8.00302
    - 0.0358325
    - -0.0109980
    effort: []
    time_from_start:
      sec: 2
      nanosec: 66276828
  -
    positions:
    - 1.48031
    - -1.56985
    - 1.67086
    - -0.101813
    - 1.48029
    - -4.71132
    velocities:
    - 9.87929e-05
    - 0.00689751
    - 0.0583651
    - -0.0652546
    - 0.000307102
    - -8.04672e-05
    accelerations:
    - 0.0121117
    - 0.845611
    - 7.15537
    - -8.00000
    - 0.0376497
    - -0.00986501
    effort: []
    time_from_start:
      sec: 2
      nanosec: 166276828
  -
    positions:
    - 1.48038
    - -1.56450
    - 1.71140
    - -0.147696
    - 1.48051
    - -4.71138
    velocities:
    - 0.00123259
    - 0.118596
    - 0.746878
    - -0.865364
    - 0.00398460
    - -0.00112086
    accelerations:
    - 0.00944863
    - 1.77958
    - 6.22509
    - -8.00353
    - 0.0346383
    - -0.0117196
    effort: []
    time_from_start:
      sec: 2
      nanosec: 266276828
  -
    positions:
    - 1.48045
    - -1.55558
    - 1.75802
    - -0.203222
    - 1.48076
    - -4.71145
    velocities:
    - 0.000202048
    - 0.0271140
    - 0.126835
    - -0.153928
    - 0.000689248
    - -0.000211304
    accelerations:
    - -0.0105009
    - -1.40918
    - -6.59189
    - 8.00000
    - -0.0358218
    - 0.0109819
    effort: []
    time_from_start:
      sec: 2
      nanosec: 366276828
  -
    positions:
    - 1.48043
    - -1.55961
    - 1.74111
    - -0.182282
    - 1.48067
    - -4.71142
    velocities:
    - -0.000809215
    - -0.101259
    - -0.503911
    - 0.605089
    - -0.00272666
    - 0.000820117
    accelerations:
    - -0.0116064
    - -1.01204
    - -6.98631
    - 7.99734
    - -0.0370996
    - 0.0101980
    effort: []
    time_from_start:
      sec: 2
      nanosec: 466276828
  -
    positions:
    - 1.48032
    - -1.56960
    - 1.67634
    - -0.107541
    - 1.48032
    - -4.71133
    velocities:
    - -0.000764725
    - -0.0493982
    - -0.449257
    - 0.498594
    - -0.00235724
    - 0.000608207
    accelerations:
    - 0.0115702
    - 1.03293
    - 6.97049
    - -8.00241
    - 0.0370558
    - -0.0102413
    effort: []
    time_from_start:
      sec: 2
      nanosec: 566276828
  -
    positions:
    - 1.46943
    - -1.57480
    - 1.65169
    - -0.0777411
    - 1.46933
    - -4.68396
    velocities:
    - -0.198963
    - -0.0445007
    - 0.0356021
    - 0.00799953
    - -0.199041
    - 0.501245
    accelerations:
    - 0.0992512
    - 0.770544
    - 7.23105
    - -8.00023
    - 0.124840
    - -0.228512
    effort: []
    time_from_start:
      sec: 2
      nanosec: 666276828
  -
    positions:
    - 1.43539
    - -1.58134
    - 1.66431
    - -0.0839691
    - 1.43531
    - -4.59809
    velocities:
    - -0.498011
    - -0.0863730
    - 0.196016
    - -0.111886
    - -0.497767
    - 1.25854
    accelerations:
    - -3.14029
    - -0.270094
    - 1.53529
    - -1.27942
    - -3.13665
    - 7.99946
    effort: []
    time_from_start:
      sec: 2
      nanosec: 766276828
  -
    positions:
    - 1.37060
    - -1.59012
    - 1.69247
    - -0.103649
    - 1.37057
    - -4.43234
    velocities:
    - -0.791198
    - -0.0752854
    - 0.378505
    - -0.306805
    - -0.790344
    - 2.06102
    accelerations:
    - -2.66550
    - 0.549538
    - 2.14404
    - -2.70596
    - -2.65661
    - 7.99930
    effort: []
    time_from_start:
      sec: 2
      nanosec: 866276828
  -
    positions:
    - 1.28066
    - -1.59282
    - 1.74184
    - -0.150732
    - 1.28077
    - -4.18660
    velocities:
    - -0.983074
    - 0.0442947
    - 0.619517
    - -0.668292
    - -0.981010
    - 2.85866
    accelerations:
    - -1.11653
    - 1.89340
    - 2.69919
    - -4.59767
    - -1.10103
    - 7.99858
    effort: []
    time_from_start:
      sec: 2
      nanosec: 966276828
  -
    positions:
    - 1.18590
    - -1.57758
    - 1.81337
    - -0.237922
    - 1.18628
    - -3.87748
    velocities:
    - -0.831126
    - 0.261612
    - 0.764115
    - -1.02936
    - -0.827857
    - 3.14159
    accelerations:
    - 2.87267
    - 2.12694
    - 0.572529
    - -2.68332
    - 2.88122
    - 3.77440e-10
    effort: []
    time_from_start:
      sec: 3
      nanosec: 66276828
  -
    positions:
    - 1.11838
    - -1.54174
    - 1.89068
    - -0.351344
    - 1.11912
    - -3.56696
    velocities:
    - -0.485346
    - 0.425300
    - 0.728433
    - -1.15538
    - -0.481746
    - 2.89556
    accelerations:
    - 4.08847
    - 0.422431
    - -2.06705
    - 1.66681
    - 4.08201
    - -7.99706
    effort: []
    time_from_start:
      sec: 3
      nanosec: 166276828
  -
    positions:
    - 1.08659
    - -1.49942
    - 1.95298
    - -0.456041
    - 1.08765
    - -3.31693
    velocities:
    - -0.178968
    - 0.399966
    - 0.512359
    - -0.912403
    - -0.176231
    - 2.09556
    accelerations:
    - 2.14674
    - -0.831039
    - -2.20183
    - 3.04253
    - 2.13651
    - -7.99915
    effort: []
    time_from_start:
      sec: 3
      nanosec: 266276828
  -
    positions:
    - 1.07679
    - -1.46481
    - 1.99364
    - -0.531289
    - 1.07807
    - -3.14689
    velocities:
    - -0.0379569
    - 0.279299
    - 0.301455
    - -0.580349
    - -0.0362764
    - 1.29555
    accelerations:
    - 0.792907
    - -1.50553
    - -2.01056
    - 3.51691
    - 0.782219
    - -7.99987
    effort: []
    time_from_start:
      sec: 3
      nanosec: 366276828
  -
    positions:
    - 1.07447
    - -1.44541
    - 2.01359
    - -0.570604
    - 1.07587
    - -3.05396
    velocities:
    - -0.0714385
    - 0.0706880
    - 0.0490541
    - -0.119979
    - -0.0710776
    - 0.739650
    accelerations:
    - -2.47830
    - -3.49655
    - -4.52153
    - 7.99700
    - -2.50049
    - 0.958109
    effort: []
    time_from_start:
      sec: 3
      nanosec: 466276828
  -
    positions:
    - 1.05977
    - -1.44646
    - 2.00716
    - -0.563206
    - 1.06115
    - -2.95905
    velocities:
    - -0.186826
    - -0.0262515
    - -0.0984366
    - 0.123576
    - -0.187130
    - 1.32378
    accelerations:
    - -0.305107
    - -0.0249373
    - -0.214310
    - 0.236965
    - -0.305696
    - 8.00199
    effort: []
    time_from_start:
      sec: 3
      nanosec: 566276828
  -
    positions:
    - 1.04239
    - -1.44892
    - 1.99731
    - -0.551009
    - 1.04374
    - -2.78716
    velocities:
    - -0.130737
    - -0.0232938
    - -0.0836047
    - 0.106071
    - -0.131002
    - 2.12245
    accelerations:
    - 1.53004
    - -0.0311259
    - 0.611920
    - -0.572017
    - 1.53124
    - 7.99951
    effort: []
    time_from_start:
      sec: 3
      nanosec: 666276828
  -
    positions:
    - 1.04051
    - -1.45171
    - 1.99346
    - -0.544385
    - 1.04184
    - -2.53529
    velocities:
    - 0.133848
    - -0.0377771
    - 0.0207840
    - 0.0176995
    - 0.133758
    - 2.92348
    accelerations:
    - 3.89471
    - -0.327480
    - 1.35460
    - -1.00507
    - 3.89650
    - 7.99390
    effort: []
    time_from_start:
      sec: 3
      nanosec: 766276828
  -
    positions:
    - 1.07370
    - -1.45689
    - 2.00148
    - -0.547049
    - 1.07503
    - -2.22425
    velocities:
    - 0.525377
    - -0.0623228
    - 0.122867
    - -0.0577888
    - 0.525421
    - 3.14159
    accelerations:
    - 3.89670
    - -0.141633
    - 0.527899
    - -0.369022
    - 3.89734
    - 7.38026e-10
    effort: []
    time_from_start:
      sec: 3
      nanosec: 866276828
  -
    positions:
    - 1.13487
    - -1.46254
    - 2.01298
    - -0.552587
    - 1.13620
    - -1.94108
    velocities:
    - 0.618334
    - -0.0400963
    - 0.0879534
    - -0.0448899
    - 0.618379
    - 2.43299
    accelerations:
    - -1.80529
    - 0.522122
    - -0.734787
    - 0.201558
    - -1.80501
    - -8.00032
    effort: []
    time_from_start:
      sec: 3
      nanosec: 966276828
  -
    positions:
    - 1.19695
    - -1.46554
    - 2.01845
    - -0.554776
    - 1.19829
    - -1.73744
    velocities:
    - 0.523277
    - -0.0108270
    - 0.0260178
    - -0.0128660
    - 0.523291
    - 1.63203
    accelerations:
    - -2.44660
    - 0.320160
    - -0.430576
    - 0.0982491
    - -2.44638
    - -8.00037
    effort: []
    time_from_start:
      sec: 4
      nanosec: 66276828
  -
    positions:
    - 1.23689
    - -1.46561
    - 2.01957
    - -0.555651
    - 1.23823
    - -1.61386
    velocities:
    - 0.270923
    - 0.00469484
    - 0.00167845
    - -0.00521661
    - 0.270941
    - 0.832002
    accelerations:
    - -2.57784
    - 0.0250597
    - -0.0957509
    - 0.0593766
    - -2.57794
    - -8.00020
    effort: []
    time_from_start:
      sec: 4
      nanosec: 166276828
  -
    positions:
    - 1.25108
    - -1.46535
    - 2.01965
    - -0.555923
    - 1.25242
    - -1.57029
    velocities:
    - 0.0104223
    - 0.000186173
    - 5.82023e-05
    - -0.000199903
    - 0.0104230
    - 0.0320000
    accelerations:
    - -2.60557
    - -0.0465432
    - -0.0145506
    - 0.0499757
    - -2.60574
    - -8.00000
    effort: []
    time_from_start:
      sec: 4
      nanosec: 266276828
  -
    positions:
    - 1.25111
    - -1.46535
    - 2.01965
    - -0.555924
    - 1.25244
    - -1.57021
    velocities:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    accelerations:
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    - 0.00000
    effort: []
    time_from_start:
      sec: 4
      nanosec: 270742116
multi_dof_joint_trajectory:
  header:
    stamp:
      sec: 0
      nanosec: 0
    frame_id: ""
  joint_names: []
  points: []
```

</details>

**Comment by urfeex** [comment]
Thank you for reporting this! And thank you for providing the trajectory, that made it possible to reproduce the issue.

I can see that the trajectory you created seems to be violating the robot's limits:

![image](https://github.com/user-attachments/assets/08369e1f-eaef-4f91-8495-714c68c64682)

Speed scaling on the real robot (or in my case a URSim) scales down trajectory execution in order to stay within the robot's limits. (Note that in my plot I scaled the speed scaling factor in order to emphasize the effect, in fact it is only slowed down by around 5-6%.

If we look at the same plot (minus the speed scaling) from GZ Sim, we can see that the velocity is also limited as it hits the joint limits. But since trajectory execution isn't slowed down, as there is no scaling involved with Gazebo, the trajectory runs away from the robot which cannot follow the trajectory anymore:

![image](https://github.com/user-attachments/assets/27da3772-1649-48c4-b860-2e2cc4a07805)

This is one of the scenarios why we have implemented speed scaling in the trajectory controller for the real robot. Since that concept is not available in GZ Sim, there's not much we can do at this point. 

**TLDR:**
It seems like the trajectory you have created does not stay within the robot's limits. While the real robot can mitigate this by scaling down execution, Gazebo cannot.

---

Edit: I've attached the wontfix label to this, as I don't really see anything we can do here. However, we maybe can change some default parameters? How did you create the trajectory?

**Comment by urfeex** [comment]
I'm going to close this for now as mentioned above. If you'd like to add anything to this, please do.

**Comment by mrjogo** [comment]
Thanks for your response (just getting back to this now). I didn't realize the scaling only happened on the actual robot, so that makes sense why it succeeded there.

However, looking back through the trajectory, the largest magnitude velocity for `wrist_3_joint` is `-3.14159`, which seems to be the correct velocity limits: https://github.com/UniversalRobots/Universal_Robots_ROS2_Description/blob/16218eedc42f0597e6b262ee59299df4d8ab33a7/config/ur5e/joint_limits.yaml#L79

The one thing I DID increase is the planning `max_acceleration` for all joints from `5.0` to `8.0`. Dropping it back down does indeed get rid of the problem, although I'm a little hazy on why, besides everything's going more slowly.

The trajectory is generated by creating cartesian waypoints from an equation, then calling `moveit::core::CartesianInterpolator::computeCartesianPath()` and `trajectory_processing::TimeOptimalTrajectoryGeneration::computeTimeStamps()` on it, so I wouldn't be surprised if it has some funkiness to it.
